### ISSUE TITLE
How to run `deken` on Github Actions?

### ISSUE THREAD

**Issue by mgeier** [pure-data/deken#247]
I've tried installing it with `apt-get` like this:

https://github.com/mgeier/deken-ci-test/blob/main/.github/workflows/main.yml

But this produces an error:

```
Self-installation of Deken failed.
Please paste any errors in the bug tracker at https://github.com/pure-data/deken/issues
```

https://github.com/mgeier/deken-ci-test/runs/2509648426

What am I doing wrong?
Do I have to install something else?

**Comment by umlaeute** [comment]
good question.

it shouldn't attempt to self-install when installed via `apt`, but it seems it fails to detect that it is system-installed.

but checking the script, i think the underlying issue is here https://github.com/pure-data/deken/blob/c9cf81e563d75582a2b4e629d898ecf72c3a0141/developer/deken#L241

where it insists on an existing ${DEKEN_HOME} (but since it is system-installed, ${DEKEN_HOME} is not created...

so the quick workaround is to run this before running deken for the first time:

~~~sh
mkdir ~/.deken
~~~

**Comment by umlaeute** [comment]
btw, i know nothing about github actions, but if you can run arbitrary docker-images, you could also try [*my* official container images](https://git.iem.at/pd/deken/container_registry)


here's how my gitlab-ci job configuration looks like:
~~~yaml
deken:
  image: registry.git.iem.at/pd/deken:latest
  script:
# create a dek-file for mypackage
    - deken package --version="${CI_COMMIT_TAG#v}" mypackage
# upload the dek-file (requires DEKEN_USERNAME/DEKEN_PASSWORD to be set)
    - deken upload --no-source-error mypackage*.dek
~~~

**Comment by umlaeute** [comment]
@mgeier would you mind filing a high-priority bug with *Debian* (rather than *Ubuntu*)?

Debian/bullseye is about to be released and only release-critical bugs can now be fixed.
as i see it, the `deken` package is practically useless, unless you know about the `mkdir`workaround.

**Comment by mgeier** [comment]
Thanks for the work-around, now there is a different error, though: https://github.com/mgeier/deken-ci-test/runs/2511139249



> would you mind filing a high-priority bug with _Debian_ (rather than _Ubuntu_)?

I don't think I've ever filed a Debian bug, how does that work?



> as i see it, the `deken` package is practically useless

Maybe it's not that bad ... I was also trying the *Debian* package of `deken` on my local computer and there it works fine!

**Comment by umlaeute** [comment]
> Thanks for the work-around, now there is a different error, though: https://github.com/mgeier/deken-ci-test/runs/2511139249

hmm.
that looks like an incompatible version of `hy`.

`hy` has shown to be a rather fast-moving target, and I've found myself fixing incompat changes more often than not.
the `ubuntu-latest` image seems to be focal (20.04LTS), which ships `hy-0.18`.
On my Debian/sid system I have `hy-0.19`, which seems to work fine with `deken-0.6.0` (that is shipped with focal).

is it possible to upgrade to ubuntu/groovy?

**Comment by mgeier** [comment]
I don't think so, it seems that 20.04 is the most recent image: https://github.com/actions/virtual-environments/tree/main/images/linux

But I tried to simply pip-install version 0.19 on it, and now it works!

**Comment by umlaeute** [comment]
i'm closing this:
- the issue with the missing `${DEKEN_HOME}` has been fixed with 2e9adadd5a96744f53c37e8304cf942d7f423483 and is part of deken-0.7.0
- the issue with hy-0.18 is really a packaging problem on Ubuntu, and not fixable  from here...

**Comment by mgeier** [comment]
A little update:

Ubuntu 20.04 is not available anymore on GitHub Actions.

Both 22.04 and 24.04 don't seem to have the `deken` package.

Instead of trying to install it manually, I used the Docker image:

https://github.com/SoundScapeRenderer/ssr/blob/7997e9b54f750f785184098545bac4839c6d1cb5/.github/workflows/pd-externals.yml#L135-L151

Note that installing `git` is optional, I'm only using it to get the version number from a Git tag. I had to do a few work-arounds for `git`, but without it, it should be straightforward:

```yml
    runs-on: ubuntu-latest
    container: registry.git.iem.at/pd/deken
```

**Comment by umlaeute** [comment]
> Both 22.04 and 24.04 don't seem to have the deken package.

wow, i didn't notice that it dropped out of Ubuntu (for 2 years!). it's available again on Ubuntu/25.04 (plucky).

anyhow, using the docker image should be fine (i guess; let's see what happens when everybody uses it :-))
